src/main/scala/cromwell/engine/CallExecutionActor.scala

```diff
@@ -64,22 +63,33 @@ class CallExecutionActor(backendCall: BackendCall) extends Actor with CromwellAc
     .setMultiplier(1.1)
     .build()
 
-  /** Intended for use with `Future#onComplete`, if the `Future` completes successfully apply `successFunction`
-    * to the result value.  If the `Future` is failed, log and message self to `Finish`. */
-  def ifSuccess[T](successFunction: T => Unit): Try[T] => Unit = {
-    case Success(s) => successFunction(s)
-    case Failure(t) =>
-      logger.error(t.getMessage, t)
-      self ! Finish(FailedExecutionHandle(t))
+  /**
+   * If the `work` `Future` completes successfully, perform the `onSuccess` work, otherwise schedule
+   * the execution of the `onFailure` work using an exponential backoff.
+   */
+  def withRetry(work: Future[ExecutionHandle], onSuccess: ExecutionHandle => Unit, onFailure: => Unit): Unit = {
+    work onComplete {
+      case Success(s) => onSuccess(s)
+      case Failure(e: Exception) =>
+        logger.error(e.getMessage, e)
+        scheduleWork(onFailure)
+      case Failure(throwable) =>
+        logger.error(throwable.getMessage, throwable)
+        throw throwable
+    }
   }
 
   override def receive = LoggingReceive {
     case mode: ExecutionMode =>
-      mode.execute(backendCall) onComplete ifSuccess { self ! IssuePollRequest(_) }
+      withRetry(mode.execute(backendCall),
+        onSuccess = self ! IssuePollRequest(_),
+        onFailure = self ! mode)
```

Comment:
It does look a bit weird, I might play around with invocation syntax of this more but essentially I want to pass it an expression that evaluates to a `Future[ExecutionHandle]` and two blocks.